src/test/java/seedu/address/logic/commands/SuggestCommandTest.java

```diff
@@ -32,63 +33,64 @@ public void constructor_null_throwsNullPointerException() {
 
     @Test
     public void execute_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> new SuggestCommand(Collections.emptyList()).execute(null));
+        assertThrows(NullPointerException.class, () -> new SuggestCommand(Collections.emptySet()).execute(null));
     }
 
     @Test
-    public void execute_emptyList_success() {
-        List<Client> beforeClientList = new ArrayList<>(model.getFilteredClientList());
-        SuggestCommand suggestCommand = new SuggestCommand(Collections.emptyList());
+    public void execute_emptySet_success() {
+        List<Client> beforeClientList = new ArrayList<>(model.getSortedFilteredClientList());
+        SuggestCommand suggestCommand = new SuggestCommand(Collections.emptySet());
         CommandResult expectedResult = new CommandResult(SuggestCommand.MESSAGE_SUGGEST_SUCCESS);
         assertEquals(suggestCommand.execute(model), expectedResult);
-        List<Client> afterClientList = new ArrayList<>(model.getFilteredClientList());
+        List<Client> afterClientList = new ArrayList<>(model.getSortedFilteredClientList());
         assertEquals(beforeClientList, afterClientList);
     }
 
     @Test
-    public void execute_listPredicatesAllTrue_noChangeToClientList() {
-        List<Client> beforeClientList = new ArrayList<>(model.getFilteredClientList());
-        List<Predicate<Client>> predicateList = Arrays.asList(c -> true, c -> true);
-        SuggestCommand suggestCommand = new SuggestCommand(predicateList);
+    public void execute_filterPredicates_correctClientList() {
+        Model newModel = new ModelManager();
+        Client client1 = new ClientBuilder().withName("client1")
+                .withContractExpiryDate(ContractExpiryDate.NULL_DATE).build();
+        Client client2 = new ClientBuilder().withName("client2")
+                .withContractExpiryDate("2-3-2020").withLastModifiedInstant("2020-10-11T15:18:35.617617Z").build();
+        Client client3 = new ClientBuilder().withName("client3")
+                .withContractExpiryDate("1-3-2020").withLastModifiedInstant("2020-10-10T16:18:35.617617Z").build();
+        Client client4 = new ClientBuilder().withName("client4")
+                .withContractExpiryDate("10-3-2020").withLastModifiedInstant("2020-10-11T15:18:35.617617Z").build();
+        newModel.addClient(client1);
+        newModel.addClient(client2);
+        newModel.addClient(client3);
+        newModel.addClient(client4);
+        assertEquals(newModel.getSortedFilteredClientList().size(), 4);
+
+        LinkedHashSet<SuggestionType> suggestionTypes = new LinkedHashSet<>();
```

Comment:
```suggestion
        Set<SuggestionType> suggestionTypes = new LinkedHashSet<>();
```

Consistent with below

Reply:
done